### PR TITLE
[Databricks]: reuse getTableDDL impl from Hive

### PR DESCRIPTION
**Pull request by mixam24**
This PR adds the following functionality to improve users´ experience while working Databricks:
* reuse `getTableDDL` implementation from Hive to get complete DDL script with location, properties, etc.
* adds `DatabricksSQLDialect` with list of functions and special keywords supported by Databricks to enable highlight in the SQL Editor
* adds native DatabricksJDBC classname to the `plugins.xml`

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/model/DatabricksMetaModel.java [hunk 1]
```diff
@@ -37,4 +52,38 @@ public DBCQueryTransformer createQueryTransformer(@NotNull DBCQueryTransformType
...
+                    while (dbResult.next()) {
+                        String line = dbResult.getString(1);
+                        if (line == null) {
+                            continue;
+                        }
+                        sql.append(line).append("\n");
+                    }
+                    String ddl = sql.toString();
+                    if (sourceObject.isView()) {
+                        ddl = ddl.replace("CREATE VIEW", "CREATE OR REPLACE VIEW");
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/model/DatabricksMetaModel.java:79]: If you read view definitions as well here, you need to add

https://github.com/dbeaver/dbeaver/blob/469ccea7637e875a98b5f9abe6f8da2720e03334/plugins/org.jkiss.dbeaver.ext.clickhouse/src/org/jkiss/dbeaver/ext/clickhouse/model/ClickhouseMetaModel.java#L162-L165

Comment (mixam24) [reply plugins/org.jkiss.dbeaver.ext.databricks/src/org/jkiss/dbeaver/ext/databricks/model/DatabricksMetaModel.java:79]: @LonwoLonwo thanks, good catch. I missed it as the driver returns both views and tables with type `TABLE`.

File: plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml [hunk 2]
```diff
@@ -3,7 +3,8 @@
 <plugin>
 
     <extension point="org.jkiss.dbeaver.generic.meta">
-        <meta id="databricks" class="org.jkiss.dbeaver.ext.databricks.model.DatabricksMetaModel" driverClass="com.simba.spark.jdbc.Driver"/>
+        <meta id="databricks-simba" class="org.jkiss.dbeaver.ext.databricks.model.DatabricksMetaModel" driverClass="com.simba.spark.jdbc.Driver" dialect="spark"/>
+        <meta id="databricks-native" class="org.jkiss.dbeaver.ext.databricks.model.DatabricksMetaModel" driverClass="com.databricks.client.jdbc.Driver" dialect="spark"/>
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml]: Please do it inline

https://github.com/dbeaver/dbeaver/blob/5463637ae1ad5764dc4ebaec4f3184c10da9faca/plugins/org.jkiss.dbeaver.ext.mssql/plugin.xml#L22-L25

**Review by LonwoLonwo — APPROVED** [review]

**Comment by LonwoLonwo** [comment]
New day, new question.

DBeaver does not have official support for this driver: `com.databricks.client.jdbc.Driver`

Can you add it? Like `com.simba.spark.jdbc.Driver`

**Comment by mixam24** [comment]
> New day, new question.
> 
> DBeaver does not have official support for this driver: `com.databricks.client.jdbc.Driver`
> 
> Can you add it? Like `com.simba.spark.jdbc.Driver`

@LonwoLonwo please take a look, not sure if I got you right.

File: plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml [hunk 3]
```diff
@@ -31,7 +31,29 @@
...
+                        iconBig="platform:/plugin/org.jkiss.dbeaver.ext.generic/icons/databricks_icon_big.png"
+                        class="com.databricks.client.jdbc.Driver"
+                        webURL="https://www.databricks.com/spark/jdbc-drivers-download"
+                        description="Azure Databricks driver"
+                        categories="sql,analytic,cloud">
+                    <replace provider="generic" driver="azure_databricks"/>
+
+                    <file type="jar" path="maven:/com.databricks:databricks-jdbc:RELEASE[2.6.32]"
+                          bundle="!drivers.databricks"/>
+                    <file type="jar" path="drivers/databricks" bundle="drivers.databricks"/>
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml]: <file type="jar" path="drivers/databricks" bundle="drivers.databricks"/>

We have different jar files there. Please remove this string.

File: plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml [hunk 4]
```diff
@@ -31,7 +31,29 @@
...
+                    <file type="license" path="https://www.databricks.com/jdbc-odbc-driver-license"
+                          bundle="drivers.databricks"/>
+                    <file type="license" path="drivers/databricks/EULA.txt" bundle="drivers.databricks"/>
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml]:
```suggestion
                    
```

Please remove license strings as well. I do not see a convenient way to use https://www.databricks.com/legal/jdbc-odbc-driver-license here

File: plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml [hunk 5]
```diff
@@ -31,7 +31,29 @@
...
+                        class="com.databricks.client.jdbc.Driver"
+                        webURL="https://www.databricks.com/spark/jdbc-drivers-download"
+                        description="Azure Databricks driver"
+                        categories="sql,analytic,cloud">
+                    <replace provider="generic" driver="azure_databricks"/>
+
+                    <file type="jar" path="maven:/com.databricks:databricks-jdbc:RELEASE[2.6.32]"
+                          bundle="!drivers.databricks"/>
+                    <file type="jar" path="drivers/databricks" bundle="drivers.databricks"/>
+                    <file type="license" path="https://www.databricks.com/jdbc-odbc-driver-license"
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml]: You need to use a path to a .txt file. Better to remove this string.

File: plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml [hunk 6]
```diff
@@ -19,8 +19,8 @@
             <drivers managable="true">
 
                 <driver
-                        id="databricks"
-                        label="Azure Databricks"
+                        id="databricks-simba"
```
Comment (LonwoLonwo) [inline plugins/org.jkiss.dbeaver.ext.databricks/plugin.xml]:
```suggestion
                        id="databricks"
```

Please return the original id value.

**Review by LonwoLonwo — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Pull request merged**